.github/workflows/preDeploy.yml

```diff
@@ -2,37 +2,46 @@ name: Process new code merged to main
 
 on:
   push:
-    branches: [main]
+    branches:
+      - main
 
 jobs:
   lint:
-    uses: Expensify/App/.github/workflows/lint.yml@main
+    runs-on: ubuntu-latest
+    steps:
+      - name: Run lint workflow
+        uses: Expensify/App/.github/workflows/lint.yml@main
 
   test:
-    uses: Expensify/App/.github/workflows/test.yml@main
+    runs-on: ubuntu-latest
+    steps:
+      - name: Run test workflow
+        uses: Expensify/App/.github/workflows/test.yml@main
 
   confirmPassingBuild:
     runs-on: ubuntu-latest
     needs: [lint, test]
     if: ${{ always() }}
 
     steps:
-      - if: ${{ needs.lint.result == 'failure' || needs.test.result == 'failure' }}
+      - name: Announce failed workflow in Slack
+        if: ${{ needs.lint.result == 'failure' || needs.test.result == 'failure' }}
         uses: Expensify/App/.github/actions/composite/announceFailedWorkflowInSlack@main
         with:
           SLACK_WEBHOOK: ${{ secrets.SLACK_WEBHOOK }}
 
-      - if: ${{ needs.lint.result == 'failure' || needs.test.result == 'failure' }}
+      - name: Exit failed workflow
+        if: ${{ needs.lint.result == 'failure' || needs.test.result == 'failure' }}
         run: exit 1
 
   chooseDeployActions:
     runs-on: ubuntu-latest
     needs: confirmPassingBuild
     outputs:
       MERGED_PR: ${{ steps.getMergedPullRequest.outputs.number }}
-      IS_AUTOMATED_PR: ${{ steps.isAutomatedPullRequest.outputs.IS_AUTOMATED_PR }}
-      SHOULD_DEPLOY: ${{ steps.shouldDeploy.outputs.SHOULD_DEPLOY }}
-      SHOULD_CP: ${{ steps.isStagingDeployLocked.outputs.IS_LOCKED && steps.hasCherryPickLabel.outputs.HAS_CP_LABEL }}
+      IS_AUTOMATED_PR: ${{ fromJSON(steps.isAutomatedPullRequest.outputs.IS_AUTOMATED_PR) }}
```

Comment:
If I remember correctly, it was because it resolved to `'true'` or `'false'` instead of `true` or `false`, which meant that as long it had any value it was always truthy (as a non-empty string), so I assumed it was a bug